### PR TITLE
git-comparison: add more diff commands

### PR DESCRIPTION
**Pull request by mlcui-google**
`jj diff` works a bit differently to `git diff`, so add more examples of `jj diff` using `--from`.

### CODE REVIEW

File: docs/git-comparison.md [hunk 1]
```diff
@@ -127,13 +127,24 @@ parent.
     <tr>
       <td>Show diff of the current change</td>
       <td><code>jj diff</code></td>
-      <td><code>git diff HEAD</code></td>
+      <td><code>git diff</code></td>
```
Comment (ilyagr) [inline docs/git-comparison.md:130]: This only works if the staging area is empty. I think you should either revert this change or add `(assuming empty staging area)`.

File: docs/git-comparison.md [hunk 2]
```diff
@@ -127,13 +127,24 @@ parent.
...
+    </tr>
+    <tr>
+      <td>Show diff from change A to change B</td>
+      <td><code>jj diff --from A --to B</code></td>
+      <td><code>git diff A B </code></td>
+    </tr>
+    <tr>
     <tr>
       <td>Show description and diff of a change</td>
       <td><code>jj show &lt;revision&gt;</code></td>
```
Comment (ilyagr) [inline docs/git-comparison.md:150]: This is unrelated to your change, but I think the git version of "Remove file in the current change" should be `git rm file`, right?

File: docs/git-comparison.md (hunk 1 above)
Comment (mlcui-google) [reply docs/git-comparison.md]: Ah, I didn't know that! Done.

File: docs/git-comparison.md (hunk 2 above)
Comment (mlcui-google) [reply docs/git-comparison.md:150]: Yes. Will submit another PR for this.

Comment (ilyagr) [reply docs/git-comparison.md:150]: Sounds good!

**Review by ilyagr — APPROVED** [review]
LGTM, thank you very much!

File: docs/git-comparison.md (hunk 2 above)
Comment (mlcui-google) [reply docs/git-comparison.md:150]: Done in #1848.

File: docs/git-comparison.md [hunk 3]
```diff
@@ -134,6 +134,17 @@ parent.
       <td><code>jj diff -r &lt;revision&gt;</code></td>
       <td><code>git diff &lt;revision&gt;^ &lt;revision&gt;</code></td>
     </tr>
+    <tr>
+      <td>Show diff from one change to the current change</td>
```
Comment (ilyagr) [inline docs/git-comparison.md:138]: This could say "another change" instead of "one change".

Comment (mlcui-google) [reply docs/git-comparison.md]: Done.
